lib/OpenQA/WebAPI.pm

```diff
@@ -395,26 +395,28 @@ sub startup {
     $api_ra->delete('job_templates/:job_template_id')->to('job_template#destroy');
 
     # api/v1/comments
-    $api_public_r->get('/jobs/:job_id/comments')->name('apiv1_list_comments')->to('comment#list');
-    $api_public_r->get('/jobs/:job_id/comments/:comment_id')->name('apiv1_get_comment')->to('comment#text');
-    $api_ru->post('/jobs/:job_id/comments')->name('apiv1_post_comment')->to('comment#create');
-    $api_ru->put('/jobs/:job_id/comments/:comment_id')->name('apiv1_put_comment')->to('comment#update');
-    $api_ra->delete('/jobs/:job_id/comments/:comment_id')->name('apiv1_delete_comment')->to('comment#delete');
-    $api_public_r->get('/groups/:group_id/comments')->name('apiv1_list_group_comment')->to('comment#list');
-    $api_public_r->get('/groups/:group_id/comments/:comment_id')->name('apiv1_get_group_comment')->to('comment#text');
-    $api_ru->post('/groups/:group_id/comments')->name('apiv1_post_group_comment')->to('comment#create');
-    $api_ru->put('/groups/:group_id/comments/:comment_id')->name('apiv1_put_group_comment')->to('comment#update');
-    $api_ra->delete('/groups/:group_id/comments/:comment_id')->name('apiv1_delete_group_comment')->to('comment#delete');
-    $api_public_r->get('/parent_groups/:parent_group_id/comments')->name('apiv1_list_parent_group_comment')
-      ->to('comment#list');
-    $api_public_r->get('/parent_groups/:parent_group_id/comments/:comment_id')->name('apiv1_get_parent_group_comment')
+    $api_public_r->get('/jobs/<job_id:num>/comments')->name('apiv1_list_comments')->to('comment#list');
+    $api_public_r->get('/jobs/<job_id:num>/comments/:comment_id')->name('apiv1_get_comment')->to('comment#text');
```

Comment:
I think it would make sense to use `:num` for the `comment_id` as well. Same for the other occurrences of course.